letshelp-letsencrypt/letshelp_letsencrypt/letshelp_letsencrypt_apache.py

```diff
@@ -0,0 +1,303 @@
+import subprocess
+import sys
+import tarfile
+import tempfile
+import textwrap
+
+
+_DESCRIPTION = """
+Let's Help is a simple script you can run to help out the Let's Encrypt
+project. Since Let's Encrypt will support automatically configuring HTTPS on
+many servers, we want to test this functionality on as many configurations as
+possible. This script will create a sanitized copy of your Apache
+configuration, notifying you of the files that have been selected. If (and only
+if) you approve this selection, these files will be sent to the Let's Encrypt
+developers.
+
+"""
+
+
+_NO_APACHECTL = """
+Unable to find `apachectl` which is required for this script to work. If it is
+installed, please run this script again with the --apache-ctl command line
+argument and the path to the binary.
+
+"""
+
+
+# Keywords likely to be found in filenames of sensitive files
+_SENSITIVE_FILENAME_REGEX = re.compile(r"^(?!.*proxy_fdpass).*pass.*$|private|"
+                                       r"secret|cert|crt|key|\.pem|\.der|rsa|"
+                                       r"dsa|pw")
+
+
+def make_and_verify_selection(server_root, temp_dir):
+    """Copies server_root to temp_dir and verifies selection with the user
+
+    :param str server_root: Path to the Apache server root
+    :param str temp_dir: Path to the temporary directory to copy files to
+
+    """
+    copied_files, copied_dirs = copy_config(server_root, temp_dir)
+
+    print textwrap.fill("A secure copy of the files that have been selected "
+                        "for submission has been created under {0}. All "
+                        "comments have been removed and the files are only "
+                        "accessible by the current user. A list of the files "
+                        "that have been included is shown below. Please make "
+                        "sure that this selection does not contain private "
+                        "keys, passwords, or any other sensitive "
+                        "information.".format(temp_dir))
+    print "\nFiles:"
+    for copied_file in copied_files:
+        print copied_file
+    print "Directories (including all contained files):"
+    for copied_dir in copied_dirs:
+        print copied_dir
+
+    sys.stdout.write("\nIs it safe to submit these files? ")
+    while True:
+        ans = raw_input("(Y)es/(N)o: ").lower()
+        if ans.startswith("y"):
+            return
+        elif ans.startswith("n"):
+            sys.exit("Your files were not submitted")
+
+
+def copy_config(server_root, temp_dir):
+    """Safely copies server_root to temp_dir and returns copied files
+
+    :param str server_root: Absolute path to the Apache server root
+    :param str temp_dir: Path to the temporary directory to copy files to
+
+    :returns: List of copied files and a list of leaf directories where
+        all contained files were copied
+    :rtype: `tuple` of `list` of `str`
+
+    """
+    copied_files, copied_dirs = list(), list()
```

Comment:
Please use `[]` instead of `list()`. I have no idea why pylint didn't detect it...